src/inmanta/data/schema.py

```diff
@@ -0,0 +1,211 @@
+import logging
+import pkgutil
+from types import ModuleType
+from typing import List, Optional
+
+import asyncpg
+from asyncpg import UndefinedTableError
+
+LOGGER = logging.getLogger(__name__)
+
+LEGACY_SCHEMA_VERSION_TABLE = "schemaversion"
+SCHEMA_VERSION_TABLE = "schemamanager"
+
+# no version table found
+VERSION_NONE = None
+# Legacy version table is present but empty (unlikely)
+VERSION_LEGACY = 0
+
+CORE_NAME = "core"
+
+create_schemamanager = """
+-- Table: public.schemaversion
+CREATE TABLE IF NOT EXISTS public.schemamanager (
+    name varchar PRIMARY KEY,
+    current_version integer NOT NULL
+);
+"""
+
+
+class Version(object):
+    """ Internal representation of a version """
+
+    def __init__(self, name, function):
+        self.name = name
+        self.function = function
+        self.version = self.parse(name)
+
+    @classmethod
+    def parse(cls, name: str) -> int:
+        return int(name[1:])
+
+
+class DBSchema(object):
+    """
+    Schema Manager, ensures the schema is up to date.
+
+    """
+
+    def __init__(self, name: str, package: ModuleType, connection: asyncpg.Connection) -> None:
+        """
+
+        :param name: unique name for this schema, best equal to extension name and used as prefix for all table names
```

Comment:
We should update the names of all existing tables as such that they are prefixed with "core"

Reply:
I prefer to not rename all tables and exempt core from the rules